packages/in_app_purchase/in_app_purchase_android/lib/src/in_app_purchase_android_platform.dart

```diff
@@ -0,0 +1,327 @@
+
+/// [IAPError.code] code used when a consuming a purchased item fails.
+const String kConsumptionFailedErrorCode = 'consume_purchase_failed';
+
+/// [IAPError.code] code used when a query for previouys transaction has failed.
+const String kRestoredPurchaseErrorCode = 'restore_transactions_failed';
+
+/// Indicates store front is Google Play
+const String kIAPSource = 'google_play';
+
+/// An [InAppPurchasePlatform] that wraps Android BillingClient.
+///
+/// This translates various `BillingClient` calls and responses into the
+/// generic plugin API.
+class InAppPurchaseAndroidPlatform extends InAppPurchasePlatform
+    with WidgetsBindingObserver {
+  InAppPurchaseAndroidPlatform._()
+      : billingClient =
+            BillingClient((PurchasesResultWrapper resultWrapper) async {
+          _purchaseUpdatedController
+              .add(await _getPurchaseDetailsFromResult(resultWrapper));
+        }) {
+    if (InAppPurchaseAndroidPlatform.enablePendingPurchase) {
+      billingClient.enablePendingPurchases();
+    }
+
+    _readyFuture = _connect();
+    WidgetsBinding.instance!.addObserver(this);
+    _purchaseUpdatedController = StreamController.broadcast();
+    ;
+  }
+
+  /// Returns the singleton instance of the [InAppPurchaseAndroidPlatform].
+  static InAppPurchaseAndroidPlatform get instance => _getOrCreateInstance();
+  static InAppPurchaseAndroidPlatform? _instance;
+
+  /// Whether pending purchase is enabled.
+  ///
+  /// See also [enablePendingPurchases] for more on pending purchases.
+  static bool get enablePendingPurchase => _enablePendingPurchase;
+  static bool _enablePendingPurchase = false;
+
+  /// Enable the [InAppPurchaseConnection] to handle pending purchases.
+  ///
+  /// This method is required to be called when initialize the application.
+  /// It is to acknowledge your application has been updated to support pending purchases.
+  /// See [Support pending transactions](https://developer.android.com/google/play/billing/billing_library_overview#pending)
+  /// for more details.
+  /// Failure to call this method before access [instance] will throw an exception.
+  static void enablePendingPurchases() {
+    _enablePendingPurchase = true;
+  }
+
+  Stream<List<PurchaseDetails>> get purchaseStream =>
+      _purchaseUpdatedController.stream;
+  static late StreamController<List<PurchaseDetails>>
+      _purchaseUpdatedController;
+
+  /// The [BillingClient] that's abstracted by [GooglePlayConnection].
+  ///
+  /// This field should not be used out of test code.
+  @visibleForTesting
+  late final BillingClient billingClient;
+
+  late Future<void> _readyFuture;
+  static Set<String> _productIdsToConsume = Set<String>();
+
+  @override
+  Future<bool> isAvailable() async {
+    await _readyFuture;
+    return billingClient.isReady();
+  }
+
+  @override
+  Future<bool> buyNonConsumable({required PurchaseParam purchaseParam}) async {
+    if (!(purchaseParam is GooglePlayPurchaseParam)) {
```

Comment:
Similar to iOS, I think we don't have to force it here. Instead, we can document that if the client app wants to use the change subscription feature, they should use a `GooglePlayPurchaseParam` instead. Otherwise, `changeSubscriptionParam` will default to `null`

Reply:
Agreed, I updated the code to reflect this behaviour.